gcc/rust/hir/rust-ast-lower-base.cc

```diff
@@ -788,14 +788,13 @@ ASTLoweringBase::handle_lang_item_attribute (const ItemWrapper &item,
 {
   auto &literal = static_cast<AST::AttrInputLiteral &> (attr.get_attr_input ());
   const auto &lang_item_type_str = literal.get_literal ().as_string ();
-  auto lang_item_type = Analysis::RustLangItem::Parse (lang_item_type_str);
-  if (lang_item_type == Analysis::RustLangItem::ItemType::UNKNOWN)
-    {
-      rust_error_at (attr.get_locus (), "unknown lang item");
-      return;
-    }
-  mappings->insert_lang_item (lang_item_type,
-			      item.get_mappings ().get_defid ());
+  auto lang_item_type = LangItem::Parse (lang_item_type_str);
+
+  if (lang_item_type)
```

Comment:
I'd prefer we keep the condition explicit instead of relying on the underlying value.

Reply:
No sorry, I ready too fast and thought we kept the Unknown item kind. Given it is an optional we can keep this.